### PR TITLE
Added `SmoothImplicit`

### PR DESCRIPTION
**Pull request by dipterix**
Hi @zarquon42b 

May I propose this PR to:

* Add implementation for `SmoothImplicit`
* Edit the documentation to comply with the latest changes in `roxygen2` and remove the warnings
* Add github action (in folder `.github`), allowing Github Action to automatically check the R package (including the future PR); see [example](https://github.com/dipterix/Rvcg/actions/runs/7641443432/job/20818774804)
* Add `*.so` to .gitignore to avoid including the compiled files into repo

TODO: Please bump the version number in `DESCRIPTION` file

### CODE REVIEW

**Comment by zarquon42b** [comment]
Thanks a lot. I'll look into it ASAP

**Comment by dipterix** [comment]
Also I noticed 

1. The `Eigen` folder in `vcglib` is not doing anything. I think I can help reduce the package size (removing unused files)
2. `vcglib` uses lots of `assert`, which causes R sessions to quit/crash. I wonder do you have any suggestions?

**Comment by zarquon42b** [comment]
1. It is there but will not be included in the R-package (https://github.com/zarquon42b/Rvcg/blob/master/.Rbuildignore). The whole vcglib folder is a git repository that I patch. If I would remove anything, I could not update it from the vcglib source when needed.
2. There is nothing we can do about it. However, IMHO they are not called other wise the CRAN checks would throw a warning that assert statements are included in the compiled code

**Comment by zarquon42b** [comment]
Hi,
I just realized that I had forgotten to push after the last release. Could you please fix the very minute conflicts in the DESCRIPTION file and remove the #export tags (and rerun roxygen)  as mentioned above. Then I can merge it into this repo.

**Review by zarquon42b — COMMENTED** [review]
I would not want to export the applyTransform functions as they are already contained in my other package Morpho. As Rvcg is a dependency of Morpho, I wanted to avoid cross-dependency and added those functions as private.

**Comment by zarquon42b** [comment]
I guess you couldn't see my remarks as the review was still in progress. Sorry for the confusion.

File: R/vcgIsosurface.r [hunk 1]
```diff
@@ -91,6 +91,7 @@ invertFaces <- function (mesh) {
 
 applyTransform <- function(x,trafo,inverse)UseMethod("applyTransform")
 
+#' @export
```
[the remark this answers is not in the document]
Comment (dipterix) [reply R/vcgIsosurface.r:94]: I see that `Morpho` imports `Rvcg`. How about re-export this function? I think it not only solves the dependency issue, but also is recommended by CRAN.

For example, in `Morpho`, you can have 

```r
#' @importFrom Rvcg applyTransform
#' @export
Rvcg::applyTransform
```

roxygen2 will mark this function as reexport, and you won't have conflict issue when loading.

All under the condition that two functions are identically implemented.

**Pull request merged**